### ISSUE TITLE
Training of SECOND on custom dataset

### ISSUE THREAD

**Issue by ste-wit** [open-mmlab/mmdetection3d#1862]
I am currently training SECOND on a custom dataset. I use the KITTI configuration as a basis for my own config.

Now when I train with the KITTI point cloud range I get non-zero evaluation results:
```
point_cloud_range = [0, -40, -3, 70.4, 40, 1]

2022-09-22 15:23:06,230 - mmdet - INFO - Epoch [1][650/667]	lr: 1.947e-03, eta: 3:57:42, time: 0.514, data_time: 0.011, memory: 5278, loss_cls: 0.1018, loss_bbox: 1.5896, loss_dir: 0.1385, loss: 1.8299, grad_norm: 5.5584
2022-09-22 15:23:15,079 - mmdet - INFO - Saving checkpoint at 1 epochs
2022-09-22 15:24:10,487 - mmdet - INFO - 
+------------+---------+---------+---------+---------+
| classes    | AP_0.25 | AR_0.25 | AP_0.50 | AR_0.50 |
+------------+---------+---------+---------+---------+
| Pedestrian | 0.7134  | 0.7888  | 0.2870  | 0.4756  |
+------------+---------+---------+---------+---------+
| Overall    | 0.7134  | 0.7888  | 0.2870  | 0.4756  |
+------------+---------+---------+---------+---------+
```

However, my custom dataset requires a different z-range. Hence, I modified the point cloud z-range (**[-1.5, 2.5]** instead of **[-3, 1]**) in the config. But then I get strange validation results:
```
point_cloud_range = [0, -40, -1.5, 70.4, 40, 2.5]

2022-09-22 15:11:43,648 - mmdet - INFO - Epoch [1][650/667]	lr: 1.947e-03, eta: 4:28:19, time: 0.584, data_time: 0.012, memory: 5416, loss_cls: 0.1031, loss_bbox: 1.4842, loss_dir: 0.1383, loss: 1.7256, grad_norm: 3.8887
2022-09-22 15:11:53,664 - mmdet - INFO - Saving checkpoint at 1 epochs
2022-09-22 15:13:00,966 - mmdet - INFO - 
+------------+---------+---------+---------+---------+
| classes    | AP_0.25 | AR_0.25 | AP_0.50 | AR_0.50 |
+------------+---------+---------+---------+---------+
| Pedestrian | 0.0000  | 0.0019  | 0.0000  | 0.0000  |
+------------+---------+---------+---------+---------+
| Overall    | 0.0000  | 0.0019  | 0.0000  | 0.0000  |
+------------+---------+---------+---------+---------+
2022-09-22 15:14:19,195 - mmdet - INFO - Epoch(val) [1][167]	loss_cls: 0.7358, loss_bbox: 1.7940, loss_dir: 0.1437, loss: 2.6735
```

The really strange this is that the loss seems to be fine in both cases. I added the loss of the broken run above and you can see that is in the same range as the training loss.

I already debugged this in depth and during the run with the modified z-range, the [test-function](https://github.com/open-mmlab/mmdetection3d/blob/9556958fe1c6fe432d55a9f98781b8fdd90f4e9c/mmdet3d/apis/test.py#L40) returns the same bounding boxes, scores and labels for every frame of the validation set.

Does anybody have an idea what the reason for this behavior is?

**Comment by ste-wit** [comment]
I also saw that there are multiple closed issues that report similar issues, such as:
#1518, #1241, #1720

Unfortunately they are not providing any solution.

**Comment by ste-wit** [comment]
I found the issue. Due to the missing `PointShuffle` step in my validation pipeline and the voxel count limit (16.000 and 40.000, respectively) there is important data in the point cloud missing.

This did not surface before as the `browse_dataset.py` tool is of course using the training pipeline and here everything was fine due to the `PointShuffle` step.
